storage-proofs-porep/src/stacked/vanilla/utils.rs

```diff
@@ -4,44 +4,41 @@ use std::slice::{self, ChunksExactMut};
 /// A slice type which can be shared between threads, but must be fully managed by the caller.
 /// Any synchronization must be ensured by the caller, which is why all access is `unsafe`.
 #[derive(Debug)]
-pub struct UnsafeSlice<'a, T> {
-    // holds the data to ensure lifetime correctness
-    data: UnsafeCell<&'a mut [T]>,
```

Comment:
I'm not 100% sure about this one, but this doesn't seem right. As the comment suggests, I think we still want to have a reference to the data for correctness purpose. Hence I would just rename it to `_data` instead.

Reply:
yep, keeping this lifetime is important, to make sure it is tracked properly

Reply:
Ok. Returned lifetime and renamed `data` to `_data` in https://github.com/filecoin-project/rust-fil-proofs/pull/1607/commits/0f535fb2a78b5be7b721c9877f6450ed48756166. Could You please elaborate more on what does `make sure it is tracked properly` mean?

Reply:
The issue is that if we just use the raw pointer the compiler won't be able to track the lifetime for us. if we keep it in here we get the tracking for it.